bread/contrib/document_templates/models.py

```diff
@@ -0,0 +1,25 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from bread.contrib.reports.fields.queryfield import QuerysetField
+
+
+class DocumentTemplate(models.Model):
+    name = models.CharField(_("Name"), max_length=255)
+    model = models.ForeignKey(
+        ContentType,
+        on_delete=models.PROTECT,
+    )
+    filter = QuerysetField(_("Filter"), modelfieldname="model")
```

Comment:
What is the exact use of model and filter here? ```filter``` I did not see used. ```model``` I understand, but even there I do not see a real advantage, as the only use seems to be determining the model inside the generation function (which could also be passed via URL, as the caller knows the correct model.

Reply:
Yup, filter is only legacy code.
Would you prefer encoding the model into the URL? I thought this is tricky because we would need a system to encode and decode the name of the model into a URL.
Would an url like that be fine? `document-template/<id>/app/<appname>/model/<modelname>/object/<object_id>`
then I think I could could do `django.apps.apps.get_model(<appname>,<modelname>)`

Reply:
I think if we would do something like checks or field-suggestions for a template, then specifying the model as a field would be useful. But as for now there seem no use, so yes, in the URL would be preferable.

Regarding encoding I suggest we use the same that is used by ForeignKey fields: "appname.modelnname"
This works also with apps: ```django.apps.apps.get_model("mayapp.mymodel")``` (which is case-insensitive btw).